test/claim-nbf.test.js

```diff
@@ -196,28 +196,22 @@ describe('not before', function() {
       expect(decoded.nbf).to.equal(60);
     });
 
-    // TODO an nbf of -Infinity should fail validation
-    it('should set null "nbf" when given -Infinity', function () {
-      const token = signWithNotBefore({nbf: -Infinity});
-
-      const decoded = jwt.decode(token);
-      expect(decoded.nbf).to.be.null;
+    it('should throw exception when given -Infinity', function () {
+      expect(() => signWithNotBefore({nbf: -Infinity})).to.throw(
+        '"nbf" should be a number of seconds'
+      );
```

Comment:
After the previous comments, this might not be a surprise. This and the following tests can be moved into the `jwt.sign "nbf" claim validation` block above.